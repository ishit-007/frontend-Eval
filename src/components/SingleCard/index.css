.single-card {
  width: 600px;
  height: 1000px;
  margin-top: 20px;
}
.card-container {
  display: flex;
  align-items: center;
  justify-content: center;
}
.single-card img {
  height: 500px;
  width: 600px;
}
.single-card .timestamp {
  display: flex;
  justify-content: space-between;
}
.single-card .content h3 {
  text-align: center;
  font-family: 'Big Shoulders Display';
}
.single-card .content {
  display: flex;
  flex-direction: column;
  gap: 10px;
  height: 250px;
  padding: 5px;
  color: white;
  background-color: black;
  border-radius: 0 0 10px 10px;
}
.single-card .description {
  height: 150px;
  overflow: auto;
}
.single-card .venue {
  padding-top: 10px;
  font-weight: 800;
  height: 30px;
}
.single-card .date,
.single-card .timestamp {
  font-weight: 800;
  height: 20px;
}
.register-unregister {
  display: flex;
  justify-content: center;
  width: 100%;
}
.register-unregister button {
  font-size: larger;
  background-color: white;
  color: black;
  border: none;
  padding: 5px 10px;
}
