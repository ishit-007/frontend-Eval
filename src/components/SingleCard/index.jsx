/* eslint-disable no-unused-vars */
import './index.css';
import React from 'react';
import Header from '../Header';
import { makeRequest, GET_EVENT_BY_ID } from '../../util/makeRequest';
import { useParams } from 'react-router-dom';
import { ThemeContext } from '../../contexts/ThemeContext';

const SingleCard = () => {
  const { appTheme } = React.useContext(ThemeContext);
  const { id } = useParams();
  const [card, setCard] = React.useState({});
  React.useEffect(() => {
    makeRequest(GET_EVENT_BY_ID(id), {}).then(data => {
      setCard(data);
    });
  }, []);

  return (
    <div className="card-container">
      <div className="single-card">
        <div className="image">
          <img src={card.imgUrl} alt="" />
        </div>

        <div className="content" style={{ backgroundColor: appTheme }}>
          <h3>{card.name}</h3>
          <p className="description">{card.description}</p>
          <p className="venue">Venue:{card.venue}</p>
          <div className="timestamp">
            <p className="date">Date:{new Date(card.datetime).toLocaleDateString()}</p>
            <p className="time">Time:{new Date(card.datetime).toLocaleTimeString()}</p>
          </div>
          <div className="options"></div>
          <div className="register-unregister">
            <button className="register">Register</button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SingleCard;
