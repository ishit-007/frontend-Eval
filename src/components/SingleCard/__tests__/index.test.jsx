import SingleCard from '..';
import renderer from 'react-test-renderer';
import React from 'react';

describe('NavBar', () => {
  it('should render the Single Card Component', async () => {
    const tree = renderer.create(<SingleCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
