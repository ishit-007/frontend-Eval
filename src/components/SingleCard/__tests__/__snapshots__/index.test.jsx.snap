// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Card should render the Single Card Component 1`] = `
<div
  className="card-container"
>
  <div
    className="single-card"
  >
    <div
      className="image"
    >
      <img
        alt=""
      />
    </div>
    <div
      className="content"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <h3 />
      <p
        className="description"
      />
      <p
        className="venue"
      >
        Venue:
      </p>
      <div
        className="timestamp"
      >
        <p
          className="date"
        >
          Date:
          Invalid Date
        </p>
        <p
          className="time"
        >
          Time:
          Invalid Date
        </p>
      </div>
      <div
        className="options"
      />
      <div
        className="register-unregister"
      >
        <button
          className="register"
          style={
            Object {
              "color": undefined,
              "fontWeight": 600,
            }
          }
        >
          REGISTER
        </button>
      </div>
    </div>
  </div>
</div>
`;
